### PR TITLE
added SmartMix policy and adapted PreMix behaviour to look at source …

### PR DESCRIPTION
**Pull request by chr1sgrant**
…solution volume

### CODE REVIEW

File: microArch/driver/liquidhandling/makelhpolicy.go [hunk 1]
```diff
@@ -874,6 +907,93 @@ func GetLHPolicyForTest() (*wtype.LHPolicyRuleSet, error) {
...
+	adjustPostMix = wtype.NewLHPolicyRule("mixInto100ul")
+	adjustPostMix.AddCategoryConditionOn("LIQUIDCLASS", "SmartMix")
+	adjustPostMix.AddNumericConditionOn("TOWELLVOLUME", 100.0, 200.0)
+	adjustVol100 := AdjustPostMixVolume(wunit.NewVolume(100, "ul"))
+
+	lhpr.AddRule(adjustPostMix, adjustVol100)
+
+	adjustPostMix = wtype.NewLHPolicyRule("mixInto200ul")
+	adjustPostMix.AddCategoryConditionOn("LIQUIDCLASS", "SmartMix")
+	adjustPostMix.AddNumericConditionOn("TOWELLVOLUME", 200.0, 500.0)
```
Comment (x-anderson) [inline microArch/driver/liquidhandling/makelhpolicy.go]: What would happen in this case in the case of well volumes >500ul?

File: microArch/driver/liquidhandling/makelhpolicy.go [hunk 2]
```diff
@@ -874,6 +907,93 @@ func GetLHPolicyForTest() (*wtype.LHPolicyRuleSet, error) {
...
+
+	lhpr.AddRule(adjustPreMix, adjustPreMixVol100)
+
+	adjustPreMix = wtype.NewLHPolicyRule("premixFrom200ul")
+	adjustPreMix.AddCategoryConditionOn("LIQUIDCLASS", "PreMix")
+	adjustPreMix.AddNumericConditionOn("WELLFROMVOLUME", 200.0, 1000.0)
+
+	lhpr.AddRule(adjustPreMix, adjustPreMixVol200)
+
+	// NeedToMix
```
Comment (x-anderson) [inline microArch/driver/liquidhandling/makelhpolicy.go]: We need to deprecate the needtomix policy, and replace it with this proposed smartmix once tested

File: microArch/driver/liquidhandling/makelhpolicy.go (hunk 1 above)
Comment (chr1sgrant) [reply microArch/driver/liquidhandling/makelhpolicy.go]: good point we can increase this value

File: microArch/driver/liquidhandling/makelhpolicy.go (hunk 2 above)
Comment (chr1sgrant) [reply microArch/driver/liquidhandling/makelhpolicy.go]: Let's see after testing

**Comment by chr1sgrant** [comment]
Accuracy test looks ok

**Pull request merged**